### PR TITLE
[minor_change] Added mac attribute to the new site function

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: models/schema_site_bd.go [hunk 1]
```diff
@@ -6,11 +6,12 @@ type SiteBd struct {
 	Value map[string]interface{} `json:",omitempty"`
 }
 
-func NewSchemaSiteBd(ops, path string, anpRef map[string]interface{}, host bool) *SiteBd {
+func NewSchemaSiteBd(ops, path, mac string, anpRef map[string]interface{}, host bool) *SiteBd {
 	var externalepgMap map[string]interface{}
 	externalepgMap = map[string]interface{}{
 		"bdRef":            anpRef,
 		"hostBasedRouting": host,
+		"mac":              mac,
```
Comment (akinross) [inline models/schema_site_bd.go:14]: What happens to when payload sent is set to empty string? Did you verify no validation errors are returned?

File: models/schema_site_bd.go [hunk 2]
```diff
@@ -6,11 +6,12 @@ type SiteBd struct {
 	Value map[string]interface{} `json:",omitempty"`
 }
 
-func NewSchemaSiteBd(ops, path string, anpRef map[string]interface{}, host bool) *SiteBd {
+func NewSchemaSiteBd(ops, path, mac string, anpRef map[string]interface{}, host bool) *SiteBd {
 	var externalepgMap map[string]interface{}
 	externalepgMap = map[string]interface{}{
```
Comment (akinross) [inline models/schema_site_bd.go:11]: We can rename the variable to correct naming and initialize directly with attributes

```suggestion
	siteBdMap := map[string]interface{}{
```

Comment (akinross) [reply models/schema_site_bd.go:11]: make sure to also adjust the return value

**Review by akinross — CHANGES_REQUESTED** [review]

File: models/schema_site_bd.go (hunk 2 above)
Comment (shrsr) [reply models/schema_site_bd.go:11]: ok

File: models/schema_site_bd.go [hunk 3]
```diff
@@ -6,17 +6,20 @@ type SiteBd struct {
...
-func NewSchemaSiteBd(ops, path string, anpRef map[string]interface{}, host bool) *SiteBd {
-	var externalepgMap map[string]interface{}
-	externalepgMap = map[string]interface{}{
+func NewSchemaSiteBd(ops, path, mac string, anpRef map[string]interface{}, host bool) *SiteBd {
+	siteBdMap := map[string]interface{}{
 		"bdRef":            anpRef,
 		"hostBasedRouting": host,
 	}
 
+	if mac != "" {
```
Comment (akinross) [inline models/schema_site_bd.go:15]: should anpRef not be bdRef?

```suggestion
func NewSchemaSiteBd(ops, path, mac string, bdRef map[string]interface{}, host bool) *SiteBd {
	siteBdMap := map[string]interface{}{
		"bdRef":            bdRef,
		"hostBasedRouting": host,
	}

	if mac != "" {
```

**Review by akinross — CHANGES_REQUESTED** [review]

File: models/schema_site_bd.go (hunk 3 above)
Comment (shrsr) [reply models/schema_site_bd.go]: ok

**Review by akinross — APPROVED** [review]
LGTM

**Review by sajagana — APPROVED** [review]
LGTM!

**Pull request merged**